[APM] docs: restructure APM documentation for 7.7

bmorelli25 opened the pull request:
## Summary

This PR:

* Separates non-reference content from reference content to create a triage location for guide content as we adapt what we have into how-to guides, and begin to create more task-focussed documentation.
* Improves the TOC structure and headings by moving content from the "Using APM" bucket, and making use of table abbreviations to simplify the TOC headings.

In addition, this PR:

* Explains where supported configurations for central config are documented
* Updates service map documentation with feedback from @nehaduggal 

**Reviewers: Please visit the doc preview link and click around. Any feedback on the new structure is welcome.**

## Doc preview

~http://kibana_63280.docs-preview.app.elstc.co/guide/en/kibana/master/xpack-apm.html~
https://www.elastic.co/guide/en/kibana/current/xpack-apm.html

## Screenshots

This is the proposed content structure:

**collapsed**

<img width="326" alt="Screen Shot 2020-04-10 at 12 46 19 PM" src="https://user-images.githubusercontent.com/5618806/79018725-5baa8980-7b29-11ea-8569-2db8e7694eda.png">

**expanded**

<img width="332" alt="Screen Shot 2020-04-10 at 12 32 50 PM" src="https://user-images.githubusercontent.com/5618806/79018208-19cd1380-7b28-11ea-852c-4f6f2ba7b891.png">

## Related Issues

Closes https://github.com/elastic/kibana/issues/60462.
Closes https://github.com/elastic/apm/issues/232.
Closes https://github.com/elastic/kibana/issues/63510.
Closes https://github.com/elastic/kibana/issues/63475.

bmorelli25 committed:
docs: update jump table

bmorelli25 committed:
docs: feedback from @nehaduggal

bmorelli25 committed:
docs: update agent configuration

bmorelli25 committed:
docs: add section on how service maps work

bmorelli25 committed:
docs: initial restructure

bmorelli25 committed:
docs: clean up

bmorelli25 committed:
docs: move using under get started

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

bmorelli25 commented:
@nehaduggal See https://github.com/elastic/kibana/pull/63280/files#diff-369bdd3f2877c4bfb5c34e738980fc81 for service map updates

bmorelli25 commented:
I need to fix a link in the APM Overview before the docs preview will work.

```
12:47:14 INFO:build_docs:Bad cross-document links:
12:47:14 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/apm/get-started/master/components.html:
12:47:14 INFO:build_docs:   - en/kibana/master/apm-bottlenecks.html
```

nehaduggal commented on docs/apm/service-maps.asciidoc:
```diff
@@ -1,37 +1,50 @@
-It allows you to easily visualize data flow and high-level statistics, like average transaction duration,
-requests per minute, errors per minute, and metrics, allowing you to quickly assess the status of your services.
+A service map is a real-time visual representation of the instrumented services in your application’s architecture.
+It shows you how these services are connected, along high-level metrics like average transaction duration,
```
missing "with" - along with high-level

nehaduggal commented on docs/apm/service-maps.asciidoc:
```diff
@@ -1,37 +1,50 @@
-* Global: All services and connections are shown.
-* Service-specific: Selecting a specific service will highlight it's connections.
+* Global: All services instrumented with APM agents and the connections between them are shown.
+* Service-specific: Highlight collections for a selected service.
```
connections instead of collections

nehaduggal commented on docs/apm/service-maps.asciidoc:
```diff
@@ -1,37 +1,50 @@
+Service maps rely on distributed traces to draw connections between services.
+As {apm-overview-ref}/distribute-tracing.html[distributed tracing] is enabled out-of-the-box for supported technologies, so are service maps.
+However, if a service isn't instrumented,
+or a `traceparent` header isn't being propagated to it,
```
"or is unable to propagate a `traceparent` header" Because you can see external service and database calls and won't see things like load balancers because they themselves don't explicitly propagate the traceparent header. Maybe we can specify what things can't propagate headers explicitly - e.g. load balancers, databases, message queues.

nehaduggal reviewed (commented):
Added a few comments.

bmorelli25 commented:
@elasticmachine, run elasticsearch-ci/docs

nehaduggal commented on docs/apm/getting-started.asciidoc:
```diff
@@ -1,22 +1,45 @@
+* *Spans* contain information about a specific code path that has been executed.
+They measure from the start to end of an activity,
+and they can have a parent/child relationship with other spans.
+* *Transactions* are a special kind of span that have extra metadata associated with them.
```
Can we clearly call out that Transaction is essentially the first span for a particular service rather than call it the highest level of work?

bmorelli25 commented:
@elasticmachine, run elasticsearch-ci/docs

bmorelli25 committed:
docs: more updates

bmorelli25 commented:
Based on feedback from @nehaduggal , I’ve made the following additional changes:

* PR feedback
* Defined traces in `traces.asciidoc` and `getting-started`.asciidoc
* Moved deployment annotations to how-to section
* Renamed “advanced queries” to “query your data"
* Added an example to the “query your data” page, and reworked content
* Improved APM settings description and added links to index details in the APM Server reference

Other changes:
* More clean up
* Moves watcher error reports to how-to section

To do / questions:
* Explain how to enable ML in `machine-learning-integration.html`
* Need to figure out what to do with the distributed tracing content in `traces.asciidoc`. This is really just copied over information from the APM Overview, and it wouldn’t even be necessary if all of our documentation lived in one place. I wonder if we can turn it into task based and move it into the how-to section. Something like, “Visualize distributed traces”, and talk about the span waterfall and service maps

bmorelli25 committed:
docs: grammar

dgieselaar commented on docs/apm/apm-alerts.asciidoc:
```diff
@@ -63,8 +67,8 @@ This guide creates an alert for the `opbeans-python` service based on the follow
-and the alert will be tagged with `apm` and `service.name:opbeans-python`.
+`Error rate | opbeans-python` is automatically set as the name of the alert,
+and `apm` and `service.name:opbeans-python` are added as tags.
 Feel free to edit either of these defaults.
```
I'm late to this - but users should probably not edit the tags. We will likely use them as lookup mechanisms in the future, and they're enabled only because the Alerting UI does not offer a way for them to be disabled yet.

bmorelli25 replied:
Really good to know, thanks. I'll fix this.

dgieselaar commented on docs/apm/getting-started.asciidoc:
```diff
@@ -1,22 +1,45 @@
-image::apm/images/apm-setup.png[Installation instructions on the APM page in Kibana]
+Elastic APM captures different types of information from within instrumented applications:
 
+* *Spans* contain information about the execution of a specific code path.
```
great that we have this here now as well!

dgieselaar reviewed (approved):
Great work Brandon - one suggestion about alert tags that I should have given earlier, but other than LGTM

bmorelli25 committed:
docs: alert feedback from @dgieselaar

bmorelli25 committed:
docs: add IE11 Warning

bmorelli25 commented:
Taking care of https://github.com/elastic/kibana/issues/63510 in this PR since I already have it open. See 4594da6.

bmorelli25 committed:
docs: more updates

bmorelli25 committed:
docs: add supported versions for Agents

bmorelli25 commented:
Added https://github.com/elastic/kibana/issues/63475 in ad3c5f395.

bmorelli25 committed:
docs: update supported versions for service maps
